### ISSUE TITLE
Ally/team terminology is terrible

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#1656]
Consider a 2v2v2: A+B vs C+D vs E+F.

Everywhere else when somebody says "team", they mean e.g. A+B.
But in Recoil it means e.g. just A.

When Recoil says that some entities are allied, it can either mean
* A and B are "allied", i.e. belong to the same allyteam.
* A+B and C+D are "allied", i.e. they have a ceasefire.

This is extremely confusing for something that is so basic, especially given how events are set up (if a widget is to only count my own units, it needs to check if they are on "my team"? But I don't want to count my teammates' units?!). The "team doesn't mean team" is the first and biggest idiosyncratic caveat that every aspiring modder has to be told pretty much at the start of their experience.

----

I'm not sure if this can be approached well. If we add a replacement term then that could result in [even more confusion when there's multiple parallel terms for each concept](https://xkcd.com/927/). But maybe it's still worth trying? Here's a set of hopefully clear terms (using the setup above as an example):
* A is an _army_ (extra neat because that is how SupCom calls it).
* A and B are armies allied in an allyteam (alliance).
* AB and CD are two _ceasefired_ allyteams (alliances).

**Comment by 12345swordy** [comment]
How about:

- Friendly
- Neutral
- Hostile

**Comment by sprunk** [comment]
Oh yea neutrality is another can of worms.

* allyteams can be just ceasefired or not with each other; there is no "neutral" vs "friendly" distinction.
* ceasefired is still enemy as far as most interfaces are concerned.
* ceasefires are one-way, making a relationship somewhat complicated to describe. This is useful for singleplayer (think the starcraft mission where kerrigan gets kidnapped, or the age of mythology mission where you kill kamos where you can sail around in a pirate ship and enemies don't attack you) but probably not a good default, see #1657.

And then:
* units can individually be "neutral", regardless of their allyteam status. So you can have a neutral unit belonging to an enemy team.
* there is a Gaia allyteam. Some things like map features (trees, rocks etc) often conventionally belong to Gaia. This looks a lot like "the neutral team", especially given there is only one Gaia, but Gaia is not a neutral allyteam. It is an enemy and is not ceasefired by default.

**Comment by DracoHouston** [comment]
In another project I used these terms:
Force - a collection of units acting as one force, overseen by one or more players
Team - a collection of allied forces

I did not find it _that_ confusing that these are Teams and Alliances here, but it is kind of an odd choice.

**Comment by WatchTheFort** [comment]
Wouldn't the standard terms be _Team_ for what we call an AllyTeam, and _Player_ for what we call a Team? Unfortunately, there's nothing good for what the engine calls a Player, but thankfully StarCraft is here to the rescue with the term Archon Mode.

**Comment by sprunk** [comment]
> Wouldn't the standard terms be Team for what we call an AllyTeam

Yes, but see below.

> and Player for what we call a Team?

Definitely a better term than "team", but idk if the best (see the last paragraph of this post). But both the terms "player" and "team" already have a different meaning so would become ambiguous on a technical dev level (in addition to the intuitive level where they are already ambiguous). New terms for existing meanings (e.g. "army" in addition to "team") keeps both of them technically unambiguous. At most we could attempt to stamp "team" out over time to cleanse the army meaning away from it.

> thankfully StarCraft is here to the rescue with the term Archon Mode.

One of these documentation lines sounds more intuitive than the others:
* an Army is usually controlled by one Player; though it can be jointly controlled by any number of Players.
* a Player is usually controlled by one Player; though there is an Archon Mode in which case a Player is jointly controlled by multiple Players.
* a Player is usually controlled by one Archon; though it can be jointly controlled by any number of Archons.

> nothing good for what the engine calls a Player

Supcom calls this a Human (as opposed to an AI, which is the other thing that can control an Army). Maybe that's also better than Player in that an AI is also in some way a "player", another failing of the current terminology, but is obviously not a Human. I suppose that would make the term Player more acceptable as a replacement for Army, but there are some bad bits. For example an Army doesn't have a nick but you'd expect a Player to.

**Comment by WatchTheFort** [comment]
"A player is controlled by:
- an AI
- one human
- multiple humans who share control (aka Archon mode)"

**Comment by SethDGamre** [comment]
Will this terminology shift not make it more confusing for noobs to use the Spring wiki reference?

Recoil has come a long way

**Comment by MeSaber** [comment]
In starcraft/broodwar they use the term 'team melee' when it refers to many players control sharing. Melee in itself is default play settings.

Archon mode is a term only starcraft(2) players understand.

**Comment by DracoHouston** [comment]
Changing 'Team' to 'Army' is probably the easier way to do this. I agree that with it built to support 2players1base/team melee/archon mode changing Team to 'Player' would only confuse. Alliance/AllyTeam is already named appropriately enough.

**Comment by springjools** [comment]
Yeah, the Allyteam concept is very confusing, and has always been. The reason is that you can comshare, so you can have several human players playing for one army, hence team. But I have pointed out many times that it's more intuitive that you then share a player.

Then later the reason was always that it's too difficult to change the code, so let's leave it as it is.
